src/core/stream.c

```diff
@@ -44,6 +44,36 @@ static struct {
 	    .listener_alloc = nni_ipc_listener_alloc,
 	},
 #endif
+	{
+	    .scheme         = "mqtt-tcp",
+	    .dialer_alloc   = nni_tcp_dialer_alloc,
+	    .listener_alloc = nni_tcp_listener_alloc,
+	},
+	{
+	    .scheme         = "mqtt-tcp4",
+	    .dialer_alloc   = nni_tcp_dialer_alloc,
+	    .listener_alloc = nni_tcp_listener_alloc,
+	},
+	{
+	    .scheme         = "mqtt-tcp6",
+	    .dialer_alloc   = nni_tcp_dialer_alloc,
+	    .listener_alloc = nni_tcp_listener_alloc,
+	},
+	{
+	    .scheme         = "tls+mqtt-tcp",
+	    .dialer_alloc   = nni_tls_dialer_alloc,
+	    .listener_alloc = nni_tls_listener_alloc,
+	},
+	{
+	    .scheme         = "tls+mqtt-tcp4",
```

Comment:
I suspect you should place mqtt+ in the front of your URL.  But really I don't think you need this at all.  Your mqtt URLs are not adding any unique functionality over just tls or tcp.

Reply:
Ok, as your wish. whatever you name it ^_^

Reply:
Yeah, in retrospect, mqtt specific URLs are not needed.  See above.